src/ptr_template.cpp

```diff
@@ -560,25 +560,28 @@ class createTemplate : public PlayerScript {
         }
     }
 
-    static void AddTemplateHotbar(Player* player, uint32 index) // Someone smarter than me needs to fix this.
+    static void AddTemplateHotbar(Player* player, uint32 index)
     { //                                                    0       1      2
         QueryResult barInfo = WorldDatabase.Query("SELECT Button, Action, Type FROM mod_ptrtemplate_action WHERE (ID = {} AND RaceMask & {} AND ClassMask & {})", index, player->getRaceMask(), player->getClassMask());
-        for (uint8 j = ACTION_BUTTON_BEGIN; j <= MAX_ACTION_BUTTONS; j++) // This is supposed to go through every available action slot and remove what's there.
-        { //                                                                 This doesn't work for spells added by AddTemplateSpells.
-            player->removeActionButton(j); //                                I don't know why and I've tried everything I can think of, but nothing's worked.
-        } //                                                                 And yes, I do want the hotbar cleared for characters that don't have any hotbar data in the template.
         if (barInfo)
         {
+            for (uint8 j = ACTION_BUTTON_BEGIN; j <= MAX_ACTION_BUTTONS; j++)
+            {
+                player->removeActionButton(j);
+            }
```

Comment:
I disagree with this change. You can see with the template characters on Blizzard's Wrath Classic beta (can be seen [here](https://www.youtube.com/watch?v=iz19WxoPpvw) with the Wrath Classic Level 70 template) that they have no action data, this is why the 70/80 templates included in the module have no data for the action bars. As I said in the comment;
> And yes, I do want the hotbar cleared for characters that don't have any hotbar data in the template.

Don't know if the setup in this PR would offer that sort of hotbar clearing functionality.

Reply:
Ah, I see. I did not consider that, changing soon

Reply:
I should ask, should the SendActionButtons/SaveToDB occur outside the if statement for this removeActionButton loop to take effect?

Reply:
it should :+1: